packages/odyssey-react-mui/src/labs/TopNav.tsx

```diff
@@ -0,0 +1,459 @@
+    prop !== "odysseyDesignTokens" && prop !== "isDisabled",
+})<{
+  odysseyDesignTokens: DesignTokens;
+  isDisabled?: boolean;
+}>(({ odysseyDesignTokens, isDisabled }) => ({
+  display: "flex",
+  alignItems: "center",
+  width: "100%",
+  padding: `${odysseyDesignTokens.Spacing2} ${odysseyDesignTokens.Spacing4}`,
+  color: `${isDisabled ? odysseyDesignTokens.TypographyColorDisabled : odysseyDesignTokens.TypographyColorHeading} !important`,
+  "&:focus-visible": {
+    borderRadius: 0,
+    outlineColor: odysseyDesignTokens.FocusOutlineColorPrimary,
+    outlineStyle: odysseyDesignTokens.FocusOutlineStyle,
+    outlineWidth: odysseyDesignTokens.FocusOutlineWidthMain,
+    backgroundColor: odysseyDesignTokens.HueNeutral50,
+    textDecoration: "none",
+  },
+}));
+
+const TopNavItemContent = ({
+  id,
+  label,
+  href,
+  target,
+  onClick,
+  isDisabled,
+}: TopNavLinkItem) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const topNavItemContentKeyHandler = useCallback(
+    (event: KeyboardEvent<HTMLDivElement>) => {
+      if (event?.key === "Enter") {
+        event.preventDefault();
+        event.stopPropagation();
+        onClick?.();
+      }
+    },
+    [onClick],
+  );
+
+  const TopNavItemContent = useMemo(() => {
+    return (
+      <TopNavListItemContainer
+        id={id}
+        key={id}
+        aria-disabled={isDisabled}
+        isDisabled={isDisabled}
+        odysseyDesignTokens={odysseyDesignTokens}
+      >
+        {
+          // Use Link for nav items with links and div for disabled or non-link items
+          isDisabled ? (
+            <NavItemContentClickContainer
+              odysseyDesignTokens={odysseyDesignTokens}
+            >
+              <TopNavItemLabelContainer
+                odysseyDesignTokens={odysseyDesignTokens}
+              >
+                {label}
+              </TopNavItemLabelContainer>
+            </NavItemContentClickContainer>
+          ) : !href ? (
+            <NavItemContentClickContainer
+              odysseyDesignTokens={odysseyDesignTokens}
+              role="button"
+              tabIndex={0}
+              onClick={onClick}
+              onKeyDown={topNavItemContentKeyHandler}
+            >
+              <TopNavItemLabelContainer
+                odysseyDesignTokens={odysseyDesignTokens}
+              >
+                {label}
+              </TopNavItemLabelContainer>
+            </NavItemContentClickContainer>
+          ) : (
+            <Link href={href} target={target} onClick={onClick}>
+              <TopNavItemLabelContainer
+                odysseyDesignTokens={odysseyDesignTokens}
+              >
+                {label}
+              </TopNavItemLabelContainer>
+            </Link>
+          )
+        }
+      </TopNavListItemContainer>
+    );
+  }, [
+    id,
+    label,
+    href,
+    target,
+    onClick,
+    isDisabled,
+    topNavItemContentKeyHandler,
+    odysseyDesignTokens,
+  ]);
```

Comment:
I think this should be a separate component with typed props. It seems weird to put a whole component in `useMemo`.

It makes sense for certain situations like a `.map` where you need to add memoized click handlers, but in a lot of situations, it's better to create new components instead.